cli/stat.go

```diff
@@ -0,0 +1,255 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/cli/clibase"
+	"github.com/coder/coder/cli/clistat"
+	"github.com/coder/coder/cli/cliui"
+)
+
+func (r *RootCmd) stat() *clibase.Cmd {
+	fs := afero.NewReadOnlyFs(afero.NewOsFs())
+	defaultCols := []string{
+		"host_cpu",
+		"host_memory",
+		"home_disk",
+		"container_cpu",
+		"container_memory",
+	}
+	formatter := cliui.NewOutputFormatter(
+		cliui.TableFormat([]statsRow{}, defaultCols),
+		cliui.JSONFormat(),
+	)
+	st, err := clistat.New(clistat.WithFS(fs))
+	if err != nil {
+		panic(xerrors.Errorf("initialize workspace stats collector: %w", err))
+	}
+
+	cmd := &clibase.Cmd{
+		Use:   "stat",
+		Short: "Show resource usage for the current workspace.",
+		Children: []*clibase.Cmd{
+			r.statCPU(st, fs),
+			r.statMem(st, fs),
+			r.statDisk(st),
+		},
+		Handler: func(inv *clibase.Invocation) error {
+			var sr statsRow
+
+			// Get CPU measurements first.
+			hostErr := make(chan error)
+			containerErr := make(chan error)
```

Comment:
Avoid blocking on these due to missing consumer.

```suggestion
			hostErr := make(chan error, 1)
			containerErr := make(chan error, 1)
```

Example: On hostErr, `containerErr <- err` will be left hanging forever.